test/unit/jest.config.js

```diff
@@ -8,6 +8,8 @@ const transpiledPackageNames = glob( 'packages/*/src/index.js' ).map(
 	( fileName ) => fileName.split( '/' )[ 1 ]
 );
 
+process.env.TZ = 'America/New_York';
```

Comment:
I'd avoid this to avoid false-positives. We should run the test suite in multiple timezones to make sure it works perfectly.

Reply:
Let's use a [GH actions matrix](https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions#jobsjob_idstrategymatrix) to run this test with a few different timezones for starters. (Implementing the same at a lower level such as Jest setup, which would also be available locally, seems comparatively hard. It's probably good to have broad coverage in CI sooner rather than later -- we can always try to carry it over to a solution that also allows running tests easily against multiple timzones in local testing.)

Reply:
Done in https://github.com/WordPress/gutenberg/pull/27552.